services/logs/filter.go

```diff
@@ -0,0 +1,216 @@
+package logs
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/onflow/flow-evm-gateway/storage"
+	"math/big"
+	"slices"
+)
+
+type FilterCriteria struct {
+	Addresses []common.Address
+	Topics    [][]common.Hash
+}
+
+// todo think about creating an interface for all the filters
+// Filter interface { Match() (chan *gethTypes.Log, error) }
+
+// RangeFilter matches all the indexed logs within the range defined as
+// start and end block height. The start must be strictly smaller than end value.
```

Comment:
it's not valid to search a range of 1 block?

Reply:
I was thinking no, but I can make it so it's possible.